src/bin/sync-categories.rs

```diff
@@ -0,0 +1,48 @@
+// Sync available crate categories from `src/categories.txt`.
+// Only needs to be run for new databases or when `src/categories.txt`
+// has been updated.
+//
+// Usage:
+//      cargo run --bin sync-categories
+
+#![deny(warnings)]
+
+extern crate cargo_registry;
+extern crate postgres;
+
+use cargo_registry::env;
+
+fn main() {
+    let conn = postgres::Connection::connect(&env("DATABASE_URL")[..],
+                                             postgres::TlsMode::None).unwrap();
+    let tx = conn.transaction().unwrap();
+    sync(&tx).unwrap();
+    tx.set_commit();
+    tx.finish().unwrap();
+}
+
+fn sync(tx: &postgres::transaction::Transaction) -> postgres::Result<()> {
+    let categories = include_str!("../categories.txt");
+    let categories: Vec<_> = categories.lines().collect();
+    let insert = categories.iter()
+                           .map(|c| format!("('{}')", c))
+                           .collect::<Vec<_>>()
+                           .join(",");
+    let in_clause = categories.iter()
+                              .map(|c| format!("'{}'", c))
+                              .collect::<Vec<_>>()
+                              .join(",");
```

Comment:
Perhaps just for the sake of hygiene, but could we use rust-postgres's ability for escaping here? (rather than doing so ourselves).

I do realize though that `categories.txt` isn't user input, I figure it's just a good idea to stick with built-in escaping.